app/models/assessment.rb

```diff
@@ -122,6 +123,12 @@ def available_recommendations
     end
   end
 
+  def selected_failure_reasons_empty?(assessment)
```

Comment:
We don’t need to pass an assessment in to this method as it’s already available to us here as `self`. You can also implicitly call methods within the same class, something like:

```
sections.all? { |section| section.selected_failure_reasons.empty? }
```